src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -320,37 +399,109 @@ private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBuffe
                         }
                     });
 
-                    var document = buffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-                    var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+                    var solution = await applyFixAsync(progressTracker, cancellationToken).ConfigureAwait(true);
 
-                    var allDiagnostics = codeCleanupService.GetAllDiagnostics();
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+                    cancellationToken.ThrowIfCancellationRequested();
 
-                    var enabedDiagnosticSets = ArrayBuilder<DiagnosticSet>.GetInstance();
+                    return workspace.TryApplyChanges(solution, progressTracker);
+                }
+            }
+        }
 
-                    foreach (var diagnostic in allDiagnostics.Diagnostics)
-                    {
-                        foreach (var diagnosticId in diagnostic.DiagnosticIds)
-                        {
-                            if (context.EnabledFixIds.IsFixIdEnabled(diagnosticId))
-                            {
-                                enabedDiagnosticSets.Add(diagnostic);
-                                break;
-                            }
-                        }
-                    }
+        private async Task<Solution> FixSolutionAsync(Solution solution, FixIdContainer enabledFixIds, ProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            // FixProjectAsync reports progress for documents within a project, so we use that information to update
+            // the solution progress.
+            var progressOffset = progressTracker.CompletedItems;
+            var projectProgressTracker = new ProgressTracker((description, completedItems, totalItems) =>
+            {
+                progressTracker.Description = description;
+                progressTracker.AddItems(completedItems - progressTracker.CompletedItems);
```

Comment:
I'm not sure this is correct. When we pass the `projectProgressTracker` into `FixProjectAsync` and it adds N items this will be called with no `completedItems` which will attempt to add `-progressTracker.CompletedItems` to the solution `progressTracker`.

Reply:
Yeah it seems not correct?

Reply:
Fixed